zulipterminal/model.py

```diff
@@ -729,17 +730,55 @@ def toggle_stream_pinned_status(self, stream_id: int) -> bool:
     def is_user_subscribed_to_stream(self, stream_id: int) -> bool:
         return stream_id in self.stream_dict
 
+    def make_reduced_stream_data(self, stream: Any) -> StreamData:
+        # stream_id has been changed to id.
+        return StreamData({'name': stream['name'],
+                           'id': stream['stream_id'],
+                           'color': stream['color'],
+                           'invite_only': stream['invite_only'],
+                           'description': stream['description']})
```

Comment:
This is not a duplication, I moved it outside the method where it was previously used and added it in the class level functions so that all other methods can access it.

Reply:
Based on this commit, the other code is not removed, so it's a copy (duplication), not a move.